### PR TITLE
[MPS] fix mps failing tests

### PR DESCRIPTION
**Pull request by kashif**
fixed vae test as well as torch.generator() call.

### CODE REVIEW

File: tests/test_pipelines.py [hunk 1]
```diff
@@ -1418,7 +1418,13 @@ def test_components(self):
         text2img = StableDiffusionPipeline(**inpaint.components).to(torch_device)
 
         prompt = "A painting of a squirrel eating a burger"
-        generator = torch.Generator(device=torch_device).manual_seed(0)
+
+        # Device type MPS is not supported for torch.Generator() api.
+        if torch_device == "mps":
+            generator = torch.manual_seed(0)
```
Comment (pcuenca) [inline tests/test_pipelines.py:1424]: 👍

File: tests/test_models_vae.py [hunk 2]
```diff
@@ -106,7 +106,21 @@ def test_output_pretrained(self):
...
+                    -4.0078e-01,
+                    -3.8323e-04,
+                    -1.2681e-01,
+                    -1.1462e-01,
+                    2.0095e-01,
+                    1.0893e-01,
+                    -8.8247e-02,
+                    -3.0361e-01,
+                    -9.8644e-03,
+                ]
```
[the remark this answers is not in the document]
Comment (kashif) [reply tests/test_models_vae.py:121]: good question... let me investigate

[the remark this answers is not in the document]
Comment (kashif) [reply tests/test_models_vae.py:121]: i believe at least on the CI side this test was failing but we never tested it with mps device... let me check on an older version next

[the remark this answers is not in the document]
Comment (pcuenca) [reply tests/test_models_vae.py:121]: Ok, I verified in `0.3.0` and your slice is correct. I was confused thinking that I had tested the same slice in `cpu`: I had not, I had only verified on `mps` and `cuda`. Apologies!

[the remark this answers is not in the document]
Comment (kashif) [reply tests/test_models_vae.py:121]: thanks!

**Pull request merged**